app.js

```diff
@@ -87,8 +88,19 @@ io.on('connection', (socket) => {
 
   socket.on('messageToClient', async (data) => {
     // TODO: you might want to pass in more useful info such as name and avatar pic
-    const { room, email, message, chatRoomName} = data;
-    const createdOn = new Date();
+    const { room, email, message} = data;
+    const now = new Date();
+    const createdOn = {
+      year: now.getFullYear(),
+      month: now.getMonth(),
+      date: now.getDate(),
+      hour: now.getHours(),
+      minute: now.getMinutes()
+    };
+    const user = await User.findOne({email: email});
+    const moreData = {
+      userName: user.name
+    }
```

Comment:
We'll probably want to put this in a function elsewhere so we don't have so much on app.js